daemon/api_sideload_n_try.go

```diff
@@ -392,7 +392,10 @@ func readComponentInfoFromContImpl(tempPath string) (*snap.ComponentInfo, error)
 	if err != nil {
 		return nil, fmt.Errorf("cannot open container: %w", err)
 	}
-	return snap.ReadComponentInfoFromContainer(compf)
+
+	// hook information isn't loaded here, but it shouldn't be needed in this
+	// context
+	return snap.ReadComponentInfoFromContainer(compf, nil)
```

Comment:
I'm not sure of this, in the end this information is used on component installation, where we would run the install hook. Note that here the information about which snap matches the component is found in `readComponentInfo` a bit after this. Maybe `ReadComponentInfoFromContainer` should find out the matching snap if snap.Info is nil. Or have a different function that would do that and do not accept nils in `ReadComponentInfoFromContainer` .

Reply:
From what I can tell, `snapstate.InstallComponentPath` doesn't take a `snap.ComponentInfo`, instead it builds one internally with with a call to `backend.OpenComponentFile`. That one does use the `snap.Info` to fill in the hooks.

Regardless of that though, do you still think it would be best to make the change you suggested?